clients/osv/src/main/kotlin/OsvApiClient.kt

```diff
@@ -44,24 +44,41 @@ import retrofit2.http.Path
  */
 interface OsvApiClient {
     companion object {
-        const val SERVER_URL_PRODUCTION = "https://api.osv.dev"
-        const val SERVER_URL_STAGING = "https://api-staging.osv.dev"
         const val BATCH_REQUEST_MAX_SIZE = 1000
 
         val JSON = Json.Default
 
+        /**
+         * Create an OsvApiClient instance for communicating with the given [server], optionally using a pre-built
+         * OkHttp [client].
+         */
+        fun create(server: Server, client: OkHttpClient? = null): OsvApiClient =
+            create(server.url, client)
```

Comment:
If we don't want to limit the use of the library to just 2 server URLs (which I'd agree we shouldn't), then IMO it adds quite a bit boiler plate compared to the value it brings - as still values other than staging and production can be passed.

Reply:
I agree this commit is a bit "fluffy", but I somehow liked how it's done for `ClearlyDefined`. As my goal was general alignment, the alternative would have been to drop the server enum code from `ClearlyDefined`, which I liked slightly less.

PS: I'll add a new commit that hopefully adds some more value to having a server enum at least in the `ClearlyDefined` case.